src/GeneratorV2/Process/Overloaders.cs

```diff
@@ -169,6 +169,229 @@ public void WritePrologue(IndentedTextWriter writer, NameTable nameTable)
+                if (typePostfix == "f") typePostfix = "";
+                typeName = "Matrix" + columns;
+                if (matrixMatch.Groups[2].Success)
+                {
+                    rows = int.Parse(matrixMatch.Groups[2].Value);
+                    typeName += "x" + rows;
+                }
+                typeName += typePostfix;
+                vectorSize = columns * rows;
+            }
+            else if (vectorMatch.Success)
+            {
+                vectorSize = int.Parse(vectorMatch.Groups[1].Value);
+                typePostfix = vectorMatch.Groups[2].Value;
+                if (typePostfix == "f") typePostfix = "";
+                typeName = "Vector" + vectorSize + typePostfix;
+            }
+            else
+            {
+                newOverloads = null;
+                return false;
+            }
+
+            NameTable nameTable = overload.NameTable.New();
+            Parameter[] singleParameters = overload.InputParameters.ToArray();
+            Parameter[] spanParameters = overload.InputParameters.ToArray();
+            Parameter[] arrayParameters = overload.InputParameters.ToArray();
+            Parameter? countParameter = null;
+            Parameter? ptrParam = null;
+            Parameter? vectorParam = null;
+            for (var i = 0; i < singleParameters.Length; i++)
+            {
+                Parameter parameter = singleParameters[i];
+                if (parameter.Length == null) continue;
+                if (parameter.Type is CSPointer ptr && ptr.BaseType is CSType baseType)
+                {
+                    if (!_existingVectorTypes.Contains(typeName))
+                    {
+                        // Some vector methods arent actually vectors.
+                        // So we just overload them like their normal types.
+                        typeName = baseType.TypeName;
+                    }
+
+                    Constant constant;
+                    if (parameter.Length is Constant cnst)
+                    {
+                        constant = cnst;
+                    }
+                    else if (parameter.Length is BinaryOperation binary)
+                    {
+                        ParameterReference reference;
+                        if (binary.Left is ParameterReference leftRef && binary.Right is Constant rightConst)
+                        {
+                            reference = leftRef;
+                            constant = rightConst;
+                        }
+                        else if (binary.Right is ParameterReference rightRef && binary.Left is Constant leftConst)
+                        {
+                            reference = rightRef;
+                            constant = leftConst;
+                        }
+                        else continue;
+
+                        countParameter = singleParameters.First(p => p.Name == reference.ParameterName);
+                    }
```

Comment:
A comment explaining this would be nice as it's not obvious what it's for.

Reply:
Tbh, im not sure if we should check both the left and right hand operators. As currently everything in the spec has the reference as the left operator and the constant value as the right operator. But i feel like this is kind of stupid to rely on?

Reply:
Yeah, checking both feels better.